botbuilder-contrib/botbuilder-storage-mongodb
# Externalized mongo client

## Purpose

Externalize the management of the MongoDB connection / client / instantiation so that the component itself is not responsible for the connection lifetime
*

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other - Please describe:
            _______________
```

## How to Test

* Get the code

```
npm run test:unit
npm run test:integration
```

* Test the code

See sample app for changed interface

## What to Check

Verify that the following are valid

* can start bot app using new constructor

## Other Information
